core/src/utils/input-shims/input-shims.ts

```diff
@@ -77,11 +77,11 @@ export const startInputShims = (config: Config) => {
   // At this point we need to look for all of the inputs not registered yet
   // and register them.
   for (const input of inputs) {
-    registerInput(input);
+    await registerInput(input);
   }
 
-  doc.addEventListener('ionInputDidLoad', ((ev: InputEvent) => {
-    registerInput(ev.detail);
+  doc.addEventListener('ionInputDidLoad', (async (ev: InputEvent) => {
+    await registerInput(ev.detail);
```

Comment:
```suggestion
  doc.addEventListener('ionInputDidLoad', ((ev: InputEvent) => {
   registerInput(ev.detail);
```

Adding async/await here doesn't do anything since there isn't anything that gets run after `registerInput`.

Reply:
@liamdebeasi I think I messed up 😄 I tried to run your suggestion but the issue still exists. Even with my suggestion from this PR. The funny thing is, when I'm removing the following lines, the click on the inputs work as expected:

```ts
doc.addEventListener('ionInputDidLoad', ((ev: InputEvent) => {
  registerInput(ev.detail);
}) as any);
```

Can you provide some additional information to this event registration? Is it really needed? This issue has a really big impact on our customers and I'm looking for a way to fix it.

Reply:
Yes, we have an internal utility called "scroll assist" which tries to automatically scroll inputs into view when focused. This helps prevent inputs from being covered by the keyboard. In order to do so, we need to "register" each input.

There are other utilities that use this input registration, but scroll assist is the biggest one.

I tried your code reproduction in the GitHub issue, and I was only able to reproduce this in WebKit/Safari. This leads me to believe that the bug is in WebKit, not Ionic Framework. I need to spend some time isolating the issue, but we should be able to get to the bottom of things one way or another. From what I noticed, Safari does not even dispatch a click event, so adding async/await here would not resolve the issue anyways.

Reply:
Thanks for your explanation. Okay, got it.

Yes, it only happens on iOS. As I tried to find out what the issue was, it all made sense because the input shim only gets loaded on iOS. Do you already have an idea, why this worked prior v5.0.6? 🤔 

If there's anything I can do, please let me know. I'm happy to help 👍